src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -547,64 +547,35 @@ export function useSwapInputsController({
-          // we can derive the slider position directly from the entered amount.
-          if (inputKey === 'inputAmount') {
-            const inputAssetBalance = internalSelectedInputAsset.value?.maxSwappableAmount || '0';
-            if (equalWorklet(inputAssetBalance, 0)) {
-              sliderXPosition.value = withSpring(0, snappySpringConfig);
-            } else {
-              const updatedSliderPosition = clamp(Number(divWorklet(amount, inputAssetBalance)) * SLIDER_WIDTH, 0, SLIDER_WIDTH);
-              sliderXPosition.value = withSpring(updatedSliderPosition, snappySpringConfig);
-            }
-          }
-          fetchQuoteAndAssetPrices();
-        };
-
-        runOnUI(updateWorklet)();
-      } else {
-        const resetValuesToZero = () => {
-          if (isFetching.value) isFetching.value = false;
-          if (isQuoteStale.value !== 0) isQuoteStale.value = 0;
-
-          const updateWorklet = () => {
-            'worklet';
-            const keysToReset = ['inputAmount', 'inputNativeValue', 'outputAmount', 'outputNativeValue'];
-            const updatedValues = keysToReset.reduce(
-              (acc, key) => {
-                const castedKey = key as keyof typeof inputValues.value;
-                acc[castedKey] = castedKey === inputKey && preserveAmount ? inputValues.value[castedKey] : 0;
-                return acc;
-              },
-              {} as Partial<typeof inputValues.value>
-            );
-            inputValues.modify(values => {
-              return {
-                ...values,
-                ...updatedValues,
-              };
-            });
-            sliderXPosition.value = withSpring(0, snappySpringConfig);
-            isQuoteStale.value = 0;
-            quoteFetchingInterval.stop();
-          };
-
-          runOnUI(updateWorklet)();
-        };
+  const resetValuesToZeroWorklet = (inputKey?: inputKeys) => {
+    'worklet';
+    quoteFetchingInterval.stop();
+    if (isFetching.value) isFetching.value = false;
+    if (isQuoteStale.value !== 0) isQuoteStale.value = 0;
+
+    const keysToReset = ['inputAmount', 'inputNativeValue', 'outputAmount', 'outputNativeValue'];
+    const inputKeyValue = inputKey ? inputValues.value[inputKey] : 0;
+    const hasDecimal = inputKeyValue.toString().includes('.');
+    const updatedValues = keysToReset.reduce(
+      (acc, key) => {
+        const castedKey = key as keyof typeof inputValues.value;
+        acc[castedKey] = castedKey === inputKey && hasDecimal ? inputValues.value[castedKey] : 0;
+        return acc;
+      },
+      {} as Partial<typeof inputValues.value>
+    );
+
+    inputValues.modify(values => {
+      return {
+        ...values,
+        ...updatedValues,
+      };
+    });
 
-        resetValuesToZero();
-      }
-    },
-    300,
-    { leading: false, trailing: true }
-  );
+    if (inputKey) {
+      sliderXPosition.value = withSpring(0, snappySpringConfig);
+    }
+  };
```

Comment:
Im not sure I understood whats happening here, would this behave the same?
```ts

  const resetValuesToZeroWorklet = (inputKey?: inputKeys) => {
    'worklet';
    quoteFetchingInterval.stop();
    if (isFetching.value) isFetching.value = false;
    if (isQuoteStale.value !== 0) isQuoteStale.value = 0;

    const resetValues = {
      inputAmount: 0,
      inputNativeValue: 0,
      outputAmount: 0,
      outputNativeValue: 0,
    } as const;

    if (!inputKey) {
      inputValues.modify(values => ({ ...values, ...resetValues }));
      return;
    }

    const inputKeyValue = inputValues.value[inputKey];
    const hasDecimal = inputKeyValue.toString().includes('.');

    inputValues.modify(values => ({
      ...values,
      ...resetValues,
      [inputKey]: hasDecimal ? inputKeyValue : 0,
    }));

    sliderXPosition.value = withSpring(0, snappySpringConfig);
  };
  ```

Reply:
yes - this would!